src/atomic.rs

```diff
@@ -560,6 +584,37 @@ impl<T> Atomic<T> {
     }
 }
 
+impl<T> fmt::Debug for Atomic<T> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let data = self.data.load(Ordering::SeqCst);
+        let (raw, tag) = decompose_data::<T>(data);
+
+        f.debug_struct("Atomic")
+            .field("raw", &raw)
+            .field("tag", &tag)
+            .finish()
+    }
+}
+
+impl<T> fmt::Pointer for Atomic<T> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let data = self.data.load(Ordering::SeqCst);
+        let (raw, _) = decompose_data::<T>(data);
+        fmt::Pointer::fmt(&raw, f)
+    }
+}
+
+impl<T> Clone for Atomic<T> {
+    /// Returns a copy of the atomic value.
```

Comment:
Rust docs don't show comments on trait function implementations, so maybe it'd be good to put a sentence in the `Atomic` struct description stating that `clone` and formatting methods are unsychronised.

Reply:
Actually, documentation on trait methods can be overridden and rustdoc shows it as you would expect. :)

Addressing this peculiarity around `Relaxed` in clones would probably be best explained as a dedicated subsection that explains how to use `Atomic`s when you don't really need atomicity, but just convenient access to heap-allocated objects. I'll address that in a future PR. A big one involving the switch from scopes to guards is coming up anyway...

Reply:
Hm, I could swear the doc wasn't there, but you're right and I probably just missed it.